src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -325,9 +325,15 @@ public int getConnectTimeout() {
      * Labels for all pods started by the plugin
      */
     public Map<String, String> getLabels() {
-        return labels == null ? Collections.emptyMap() : labels;
+        return labels == null || labels.isEmpty() ? DEFAULT_POD_LABELS : labels;
```

Comment:
Given the current value of `DEFAULT_POD_LABELS`, the problem would be the same if you have several Jenkins instances running in the same namespace.

Maybe the `DEFAULT_POD_LABELS` should contain a reference to the current Jenkins instance (root url?) and cloud name in order to disambiguate.

Reply:
sounds like a good improvement :)